asttokens/util.py

```diff
@@ -201,12 +222,13 @@ def visit_tree(node, previsit, postvisit):
       for n in iter_children(current):
         stack.insert(ins, (n, pvalue, _PREVISIT))
     else:
-      ret = postvisit(current, par_value, value)
+      ret = postvisit(current, par_value, cast(Optional[Token], value))
   return ret
 
 
 
 def walk(node):
+  # type: (Module) -> Iterator[Module]
```

Comment:
```suggestion
  # type: (Module) -> Iterator[NodeNG]
```

Reply:
I don't think this is right, given the first item it returns is the input Module. Possibly `Union[Module, NodeNG]` but hard to tell in the full-on dynamic typing going on here.

Reply:
Most of the mentions of `NodeNG` aren't right since the functions work with either an `astroid` or `ast` node. I thought you were just using `NodeNG` to mean `ast.AST` for some reason. But this definitely yields more than just modules.

Reply:
I've just ran some experiments on this one and found the correct set of types was `Union[Module, AST, NodeNG]`

Are you saying that most of the `NodeNG` refs should actually be `Union[AST, NodeNG]`?

Reply:
Sorry I forgot about this. Yes, technically, I think they should usually be `Union[AST, NodeNG]`, i.e. they work with both `ast` and `astroid`.

Reply:
I've added a `AstNode` type alias for `Union[AST, NodeNG]` and moved over most of the util usages of `NodeNG` to that. There are two exceptions, because they use features only on `NodeNG`